### PR TITLE
Use lowercase maintenance class names

### PR DESCRIPTION
**Pull request by paladox**
## Summary by CodeRabbit

- **New Features**
	- Enhanced command-line options for data rebuilding processes, providing more granular control.
	- Improved auto-recovery feature for interrupted rebuild processes.

- **Bug Fixes**
	- Updated error handling for unknown namespaces in command-line options.

- **Tests**
	- Updated test method names for clarity and consistency.
	- Modified instantiation of maintenance runners to align with updated naming conventions.

- **Chores**
	- Refined autoloader configuration for class names, ensuring consistency in case sensitivity.

### CODE REVIEW

**Comment by paladox** [comment]
@coderabbitai full review

**Comment by paladox** [comment]
Ok this looks like it works now.

**Comment by paladox** [comment]
@JeroenDeDauw could you review/merge this please? :)

**Comment by JeroenDeDauw** [comment]
@paladox are you aware of github.dev editor? You can get there by hitting `.` while on the SMW repo, or by going to https://github.dev/SemanticMediaWiki/SemanticMediaWiki. I never use it, but I imagine it might be a better web editing experience for if you need  change multiple files.

**Comment by paladox** [comment]
> https://github.dev/SemanticMediaWiki/SemanticMediaWiki

Oh, no, I never heard of this. Thanks!

**Comment by alistair3149** [comment]
@paladox is the PR ready for review?

File: tests/bootstrap.php
Comment (alistair3149) [inline tests/bootstrap.php:27]: Should be `SMW\Maintenance\dumpRDF`

File: tests/phpunit/Integration/Maintenance/RunImportTest.php
Comment (alistair3149) [inline tests/phpunit/Integration/Maintenance/RunImportTest.php:41]: Needs to update the class in `src\MediaWiki\Hooks.php` as well

**Review by alistair3149 — CHANGES_REQUESTED** [review]
There are a few missing class names, but otherwise LGTM!

File: tests/bootstrap.php
Comment (paladox) [reply tests/bootstrap.php]: Thanks, done.

File: tests/phpunit/Integration/Maintenance/RunImportTest.php
Comment (paladox) [reply tests/phpunit/Integration/Maintenance/RunImportTest.php:41]: Thanks, done.

**Comment by alistair3149** [comment]
@coderabbitai full review

File: tests/phpunit/Integration/MediaWiki/Import/Maintenance/RebuildFulltextSearchTableTest.php
Comment (alistair3149) [reply tests/phpunit/Integration/MediaWiki/Import/Maintenance/RebuildFulltextSearchTableTest.php:53]: For this I probably need a second opinion from @JeroenDeDauw , I am not sure how shall we handle the class names for tests.

Comment (paladox) [reply tests/phpunit/Integration/MediaWiki/Import/Maintenance/RebuildFulltextSearchTableTest.php:53]: From what I can tell it does new $class_name here https://github.com/SemanticMediaWiki/SemanticMediaWiki/blob/master/tests/phpunit/Utils/Runners/MaintenanceRunner.php#L83

Comment (alistair3149) [reply tests/phpunit/Integration/MediaWiki/Import/Maintenance/RebuildFulltextSearchTableTest.php:53]: It is called correctly. The part that I'm unsure about is whether it is okay to have the inconsistency, so I would probably defer to someone who's more knowledge on that issue.

**Review by alistair3149 — APPROVED** [review]
LGTM but someone else should take a look

File: tests/phpunit/Integration/MediaWiki/Import/Maintenance/RebuildFulltextSearchTableTest.php
Comment (JeroenDeDauw) [reply tests/phpunit/Integration/MediaWiki/Import/Maintenance/RebuildFulltextSearchTableTest.php:53]: Consistency is nice but not all-important. @paladox what is the point of this PR? Does it fix a compatibility issue somehow?

Comment (paladox) [reply tests/phpunit/Integration/MediaWiki/Import/Maintenance/RebuildFulltextSearchTableTest.php:53]: The tests were using the wrong casing for the maintenance scripts and was caught in the test by failing. Dunno why it passes in 1.39/40 but I think I saw this failure in the 1.42 test.

Comment (paladox) [reply tests/phpunit/Integration/MediaWiki/Import/Maintenance/RebuildFulltextSearchTableTest.php:53]: src/Maintenance/MaintenanceFactory.php Uses the wrong casing for the maintenance scripts which this change fixes as well. Casing matters with classes. You can’t do namespace test/test; class Test {} and then do new test/test/test(). It has to match the casing. It’s case sensitive.

**Pull request merged**
